code/modules/mob/living/carbon/xenomorph/XenoProcs.dm

```diff
@@ -264,18 +271,31 @@
 						frozen = FALSE
 						update_canmove()
 
-				if(3) //Ravagers get a free attack if they charge into someone. This will tackle if disarm is set instead.
-					var/extra_dam = min(melee_damage_lower, rand(melee_damage_lower, melee_damage_upper) * (0.3 + 0.3 * upgrade)) //About 15 to 84 extra damage depending on upgrade level.
-					M.attack_alien(src,  extra_dam) //Ancients deal about twice as much damage on a charge as a regular slash.
-					M.KnockDown(1)
+				if(RAV_CHARGE_TYPE) //Ravagers get a free attack if they charge into someone.
+					process_ravager_charge(TRUE, M)
 
 				if(4) //Predalien.
 					M.attack_alien(src) //Free hit/grab/tackle. Does not weaken, and it's just a regular slash if they choose to do that.
 
 		throwing = FALSE //Resert throwing since something was hit.
 		return TRUE
+	process_ravager_charge(FALSE)
+	return ..() //Do the parent otherwise, for turfs.
 
-	..() //Do the parent otherwise, for turfs.
+/mob/living/carbon/Xenomorph/proc/process_ravager_charge(hit = TRUE, mob/living/carbon/M = null)
+	if(!istype(src,/mob/living/carbon/Xenomorph/Ravager))
```

Comment:
Make this `return FALSE` and redefine this for `/mob/living/carbon/Xenomorph/Ravager/process_ravager_charge(hit = TRUE, mob/living/carbon/M = null)`